datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnector.java

```diff
@@ -697,6 +698,44 @@ public void testChangeDatatstreamAssignment() throws Exception {
 
   }
 
+  @Test
+  public void testGetStateForMultipleTasks() throws Exception {
+    String yummyTopic = "YummyPizza";
+    String saltyTopic = "SaltyPizza";
+    String saladTopic = "HealthySalad";
+
+    createTopic(_zkUtils, saladTopic);
+    createTopic(_zkUtils, yummyTopic);
+    createTopic(_zkUtils, saltyTopic);
+
+    // create a datastream to consume from topics ending in "Pizza"
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("pizzaStream", _broker, "\\w+Pizza");
+
+    // create three tasks for this datastream
+    List<DatastreamTask> tasks = ImmutableList.of(new DatastreamTaskImpl(Collections.singletonList(datastream)),
```

Comment:
This can be written more succinctly as:
```java
    DatastreamTaskImpl dst = new DatastreamTaskImpl(Collections.singletonList(datastream));
    List<DatastreamTask> tasks = Collections.nCopies(3, dst);
```
However, I think you intended to create three distinct objects instead. Even with multiple instantiations, `DatastreamTaskImpl::hashCode()` will return the same value for all three objects. This means the connector won't be really maintaining three independent DSTs/CTs but only a single one. You might need to alter the `id` on each DSTImpl for instance.